src/FlaUI.Core/Logging/AbstractBaseLogger.cs

```diff
@@ -0,0 +1,233 @@
+﻿using System;
+
+namespace FlaUI.Core.Logging
+{
+    public abstract class AbstractBaseLogger : ILogger
+    {
+        public bool IsTraceEnabled { get; set;}
+        public bool IsDebugEnabled { get; set; }
+        public bool IsInfoEnabled { get; set; }
+        public bool IsWarnEnabled { get; set; }
+        public bool IsErrorEnabled { get; set; }
+        public bool IsFatalEnabled { get; set; }
+
+        public abstract void GatedTrace(string message);
+        public abstract void GatedDebug(string message);
+        public abstract void GatedInfo(string message);
+        public abstract void GatedWarn(string message);
+        public abstract void GatedError(string message);
+        public abstract void GatedFatal(string message);
+
+        public AbstractBaseLogger()
+        {
+            IsTraceEnabled = false;
+            IsDebugEnabled = false;
+
+            // Default log level to info as info is the lowest
+            // log level which should not be considered 'FlaUI developer'
+            // focused.
+            IsInfoEnabled = true;
+            IsWarnEnabled = true;
+            IsErrorEnabled = true;
+            IsFatalEnabled = true;
+        }
+
+        public void Trace(string message)
+        {
+            if (IsTraceEnabled)
+            {
+                GatedTrace(message);
+            }
+        }
+
+        public void Debug(string message)
+        {
+            if (IsDebugEnabled)
+            {
+                GatedDebug(message);
+            }
+        }
+
+        public void Info(string message)
+        {
+            if (IsInfoEnabled)
+            {
+                GatedInfo(message);
+            }
+        }
+
+        public void Warn(string message)
+        {
+            if (IsWarnEnabled)
+            {
+                GatedWarn(message);
+            }
+        }
+
+        public void Error(string message)
+        {
+            if (IsErrorEnabled)
+            {
+                GatedError(message);
+            }
+        }
+
+        public void Fatal(string message)
+        {
+            if (IsFatalEnabled)
+            {
+                GatedFatal(message);
+            }
+        }
+
+        public void Trace(string message, Exception exception)
+        {
+            Trace(String.Concat(message, Environment.NewLine, exception));
+        }
+
+        public void TraceFormat(string format, params object[] args)
```

Comment:
This was just an old thing i found laying around :) We can adjust this. I can even do this after this PR.